core-meta.html

```diff
@@ -68,12 +68,54 @@
          * @type String
          * @default 'default'
          */
-        type: String
+        type: String,
+
+        /**
+         * The value of the meta-data to store. If no value is specified, the
+         * value defualts to a self reference (`this`).
+         *
+         * @attribute value
+         * @type *
+         * @default this
+         */
+        value: String,
+
+        /**
+         * The key used to store the `value` under the `type` namespace.
+         * Defaults to the `id` value.
+         *
+         * @attribute key
+         * @type String
+         * @default this.id
+         */
+        key: String
```

Comment:
We should support either `id` or `key` and not both.  You can use `attributeChanged` to listen for `id` changes.  But if you think `key` is better representing this property then I am fine with keeping `key` and remove all the `id` references.